### PR TITLE
Modify public documentation in core/include/thread

### PR DESCRIPTION
**Pull request by HalynaDumych**
Review and modify public documentation for `thread` and test its build
in Doxygen.

Relates-to: OLPEDGE-1449

Signed-off-by: Halyna Dumych <ext-halyna.dumych@here.com>

### CODE REVIEW

File: olp-cpp-sdk-core/include/olp/core/thread/TaskScheduler.h [hunk 1]
```diff
@@ -77,16 +83,16 @@ class CORE_API TaskScheduler {
...
+   * Implement this method in the subclass that takes `TaskScheduler`
+   * as a base and provides a custom algorithm for scheduling tasks
    * enqueued by the SDK.
    *
-   * @param[in] func Rvalue reference of the task to be enqueued.
-   * Move this task into your queue. No internal reference is
-   * kept, once called you own the task.
+   * @param[in] func The rvalue reference of the task that should be enqueued.
+   * Move this task into your queue. No internal references are
+   * kept. Once the task is called, you own it.
```
Comment (andescu) [inline olp-cpp-sdk-core/include/olp/core/thread/TaskScheduler.h]:
```suggestion
   * kept. Once this method is called, you own the task.
```
The ownership of the task is handed over, when `EnqueueTask` is called, not when the task is called.

File: olp-cpp-sdk-core/include/olp/core/thread/Atomic.h [hunk 2]
```diff
@@ -25,30 +25,50 @@
...
- * Depending on the provided MutexType this can be either multi-read /
- * single-write or single-read / single-write.
+ * @brief A simple atomic wrapper.
+ *
+ * Depending on the provided `MutexType`, it can be
+ * multi-read/single-write or single-read/single-write.
+ *
+ * @tparam Type The member type to which atomic access is required.
+ * @tparam MutexType Defines the lock type.
+ * @tparam ReadLockType Unlocks `MutexType`.
```
[the remark this answers is not in the document]
Comment (HalynaDumych) [reply olp-cpp-sdk-core/include/olp/core/thread/Atomic.h]: Changed

File: olp-cpp-sdk-core/include/olp/core/thread/Atomic.h [hunk 3]
```diff
@@ -25,30 +25,50 @@
...
           typename ReadLockType = std::lock_guard<MutexType> >
 class Atomic {
  public:
+  /**
+   * @brief Alias for the type.
+   */
   using value_type = Type;
+
+  /**
+   * @brief Alias for the `std::lock_guard<MutexType>` type.
```
[the remark this answers is not in the document]
Comment (HalynaDumych) [reply olp-cpp-sdk-core/include/olp/core/thread/Atomic.h]: Changed

File: olp-cpp-sdk-core/include/olp/core/thread/TaskScheduler.h (hunk 1 above)
Comment (HalynaDumych) [reply olp-cpp-sdk-core/include/olp/core/thread/TaskScheduler.h]: Changed

**Pull request merged**